beacon-chain/blockchain/fork_choice.go

```diff
@@ -148,12 +149,21 @@ func (c *ChainService) ApplyForkChoiceRule(
 	if err != nil {
 		return fmt.Errorf("could not hash head block: %v", err)
 	}
+
+	chainHead, err := c.beaconDB.ChainHead()
+	if err != nil {
+		return fmt.Errorf("could not retrieve chain head: %v", err)
+	}
+	chainRoot, err := hashutil.HashBeaconBlock(chainHead)
+	if err != nil {
+		return fmt.Errorf("could not hash current chain head: %v", err)
+	}
 	c.canonicalBlocksLock.Lock()
 	defer c.canonicalBlocksLock.Unlock()
 	c.canonicalBlocks[head.Slot] = headRoot[:]
 
 	newState := postState
-	if head.Slot != block.Slot {
+	if chainRoot != bytesutil.ToBytes32(head.ParentRootHash32) {
```

Comment:
A reorg happens when the new head does not meet these conditions
- newState != previousHead
- newState is some decedent of previousHead